src/directions.js

```diff
@@ -15,6 +16,36 @@ var Directions = L.Class.extend({
         this._waypoints = [];
     },
 
+    geocode: function(query, cb) {
+        return corslite(L.Util.template(Directions.GEOCODER_TEMPLATE, {
+            query: query,
+            token: this.options.accessToken || L.mapbox.accessToken
+        }), L.bind(function (err, resp) {
+            if (err && err.type === 'abort') {
+                return;
+            }
+
+            resp = resp || err;
```

Comment:
Here's what I suggest for the remainder of this method:

```
if (err) return cb(err);

try {
   resp = JSON.parse(resp.responseText);
} catch (e) {
    return cb(e);
}

return cb(null, resp);
```

So:

* Pass the corslite or parsing error directly to the callback, reserving `this.fire('error')` for actual direction request errors.
* Let the callback receive the full response. It can take care of extracting the first result, or showing a list of results if desired.